crates/wash-lib/src/common.rs

```diff
@@ -250,3 +256,98 @@ pub async fn get_all_inventories(
         .filter_map(Result::transpose)
         .collect::<anyhow::Result<Vec<HostInventory>>>()
 }
+
+/// Clone a git repository
+pub async fn clone_git_repo(
+    git_cmd: Option<String>,
+    tmp_dir: impl AsRef<Path>,
+    repo_url: String,
+    sub_folder: Option<String>,
+    repo_branch: Option<String>,
+) -> Result<()> {
+    let git_cmd = git_cmd.unwrap_or_else(|| DEFAULT_GIT_PATH.into());
+    let tmp_dir = tmp_dir.as_ref();
+    let cwd =
+        std::env::current_dir().map_err(|e| anyhow!("could not get current directory: {}", e))?;
+    std::env::set_current_dir(tmp_dir)
```

Comment:
Rather than mucking with the current dir, why don't we just pass `tmp_dir` as an argument to `git clone`?

Reply:
The other option, which after reading the code might be easier, is to set the cwd of the process (i.e. `cmd.current_dir`) on each of the commands

Reply:
This is a great idea, but IMO this should be handled separately. The code that does it this way has been there for a long time, and IMO how this whole function works should be changed. 

We shouldn't IMO be getting the current dir *inside* this function in the first place.

Reply:
Made an issue for this! https://github.com/wasmCloud/wasmCloud/issues/3800